apps/typegpu-docs/src/content/examples/simple/color/index.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  oklabGamutClip,
+  oklabGamutClipAlphaSlot,
+  oklabGamutClipSlot,
+  oklabToLinearRgb,
+  oklabToRgb,
+} from '@typegpu/color';
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+
+const cssProbePosition = d.vec2f(0.5, 0.5);
+
+const presentationFormat = navigator.gpu.getPreferredCanvasFormat();
+const canvas = document.querySelector('canvas') as HTMLCanvasElement;
+const cssProbe = document.querySelector('#css-probe') as HTMLDivElement;
+const probePositionText = document.querySelector(
+  '#probe-position',
+) as HTMLDivElement;
+canvas.parentElement?.appendChild(cssProbe);
+canvas.parentElement?.appendChild(probePositionText);
+const context = canvas.getContext('webgpu') as GPUCanvasContext;
+
+const cleanupController = new AbortController();
+
+document.addEventListener(
+  'mousemove',
+  (ev) => {
+    const rect = canvas.getBoundingClientRect();
+    cssProbePosition.x = (ev.clientX - rect.x) / rect.width;
+    cssProbePosition.y = 1 - (ev.clientY - rect.y) / rect.height;
+    draw();
+  },
+  {
+    signal: cleanupController.signal,
+  },
+);
+
+const root = await tgpu.init();
+
+context.configure({
+  device: root.device,
+  format: presentationFormat,
+  alphaMode: 'premultiplied',
+});
+
+const mainVertex = tgpu['~unstable'].vertexFn({
+  in: { vertexIndex: d.builtin.vertexIndex },
+  out: { outPos: d.builtin.position, uv: d.vec2f },
+}) /* wgsl */`{
+    var pos = array<vec2f, 3>(
+      vec2(-1, -1),
+      vec2(3, -1),
+      vec2(-1, 3)
+    );
+
+    return Out(vec4f(pos[in.vertexIndex], 0.0, 1.0), pos[in.vertexIndex]);
+  }`;
+
+const Uniforms = d.struct({
+  hue: d.f32,
+  alpha: d.f32,
+});
+
+const uniforms = Uniforms({
+  hue: 0.7,
+  alpha: 0.05,
+});
+
+const uniformsBindGroupLayout = tgpu.bindGroupLayout({
+  uniforms: { uniform: Uniforms },
+});
+
+const uniformsBuffer = root.createBuffer(Uniforms, uniforms).$usage('uniform');
+const uniformsBindGroup = root.createBindGroup(uniformsBindGroupLayout, {
+  uniforms: uniformsBuffer,
+});
+
+const modulo = tgpu['~unstable'].fn(
+  [d.f32, d.f32],
+  d.f32,
+)((a, b) => {
+  let m = a % b;
+  if (m < 0.0) {
+    if (b < 0.0) {
+      m -= b;
+    } else {
+      m += b;
+    }
+  }
+  return m;
+});
+
+const scaleView = tgpu['~unstable'].fn(
+  [d.vec2f],
+  d.vec2f,
+)((pos) => {
+  'kernel & js';
+  return d.vec2f(0.3 * pos.x, (pos.y * 1.2 + 1) * 0.5);
+});
+
+const patternCheckers = tgpu['~unstable'].fn(
+  [d.vec2f, d.vec3f],
+  d.f32,
+)((uv, _clipLab) => {
+  const stripeX = modulo(uv.x, 0.1) / 0.1 > 0.5;
+  const stripeY = modulo(uv.y, 0.1) / 0.1 > 0.5;
+  return d.f32(stripeX !== stripeY);
+});
```

Comment:
I remember I had a checkerboard floor in one of my apps, and I don't remember where I got it from, but here's an implementation that doesn't use `modulo`:

```ts
const patternCheckers = tgpu['~unstable'].fn(
  [d.vec2f, d.vec3f],
  d.f32,
)((uv, _clipLab) => {
  const suv = std.floor(std.mul(20, uv));
  return (suv.x + suv.y) - 2 * std.floor((suv.x + suv.y) * 0.5);
});
```

Reply:
took me a while, but this is a really smart way of doing it, no divisions.